staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/union_content.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamiccertificates
+
+import (
+	"strings"
+)
+
+type unionCAContent []CAContentProvider
+
+// NewUnionCAContentProvider returns a CAContentProvider that is a union of other CAContentProviders
+func NewUnionCAContentProvider(caContentProviders ...CAContentProvider) CAContentProvider {
+	return unionCAContent(caContentProviders)
+}
+
+// Name is just an identifier
+func (c unionCAContent) Name() string {
+	names := []string{}
+	for _, curr := range c {
+		names = append(names, curr.Name())
+	}
+	return strings.Join(names, ",")
+}
+
+// CurrentCABundleContent provides ca bundle byte content
+func (c unionCAContent) CurrentCABundleContent() []byte {
+	caBundle := []byte{}
+	for _, curr := range c {
+		caBundle = append(caBundle, curr.CurrentCABundleContent()...)
```

Comment:
I think its worth some thought about how PEMs without newlines work here. We might want to add a test for it. I believe that PEM parsing requires newlines before the `BEGIN` (except for the first one) and therefore if two pems get appended without a newline, we will miss a cert. I suspect its always safe to `bytes.Join` with a newline